autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/load/YarnLoadEvaluator.java

```diff
@@ -109,55 +115,50 @@ protected void execute() {
     }
 
     protected void pollYarnMetricsAndScaleCluster() throws Exception {
-
         StackV4Response stackV4Response = cloudbreakCommunicator.getByCrn(cluster.getStackCrn());
-
-        String hostGroupInstanceType =
-                stackResponseUtils.getHostGroupInstanceType(stackV4Response, loadAlert.getScalingPolicy().getHostGroup());
+        Map<String, String> hostFqdnsToInstanceId = stackResponseUtils.getCloudInstanceIdsForHostGroup(stackV4Response, policyHostGroup);
 
         YarnScalingServiceV1Response yarnResponse = yarnMetricsClient
-                .getYarnMetricsForCluster(cluster, hostGroupInstanceType, stackV4Response.getCloudPlatform());
+                .getYarnMetricsForCluster(cluster, policyHostGroup, hostFqdnsToInstanceId.keySet());
 
-        Map<String, String> hostFqdnsToInstanceId = stackResponseUtils
-                .getCloudInstanceIdsForHostGroup(stackV4Response, loadAlert.getScalingPolicy().getHostGroup());
         yarnResponse.getScaleUpCandidates().ifPresentOrElse(
```

Comment:
Logically Yarn should either give a ScaleUp Count or Decommission Node because both are mutually exclusive. So if scaleUp is asked, decommission nodes should be empty but since cluster size is already above max, scale up event "-2" forces it down. But one corner case is scaleUp count is given and decommission nodes contains unhealthy nodes, in this case decommissionUnhealthy nodes was given lower priority.

Now I have updated patch to make limit check more explicit for all the conditions. 

Also forced scaleDown is done only after yarnResponse so that we can collect any available yarn decommissionCandidates.